Source/DafnyCore/AST/Expressions.cs

```diff
@@ -1183,6 +1183,9 @@ public abstract class ResolverType : Type {
     public override bool ComputeMayInvolveReferences(ISet<DatatypeDecl>/*?*/ visitedDatatypes) {
       return false;
     }
+    public override Type Subst(Dictionary<TypeParameter, Type> subst) {
+      throw new NotImplementedException();
```

Comment:
Good point. Since this PR was already auto-merged, I will incorporate this change in PR https://github.com/dafny-lang/dafny/pull/3056, which had triggered this one.